m2cgen/assemblers/__init__.py

```diff
@@ -25,75 +25,128 @@
+        StatsmodelsLinearModelAssembler,
+
+    # Lightning Linear Regressors
+    "lightning.impl.adagrad.AdaGradRegressor": SklearnLinearModelAssembler,
+    "lightning.impl.primal_cd.CDRegressor": SklearnLinearModelAssembler,
+    "lightning.impl.fista.FistaRegressor": SklearnLinearModelAssembler,
+    "lightning.impl.sag.SAGARegressor": SklearnLinearModelAssembler,
+    "lightning.impl.sag.SAGRegressor": SklearnLinearModelAssembler,
+    "lightning.impl.sdca.SDCARegressor": SklearnLinearModelAssembler,
+
+    # Sklearn Linear Classifiers
+    "sklearn.linear_model.logistic.LogisticRegression":
+        SklearnLinearModelAssembler,
+    "sklearn.linear_model.logistic.LogisticRegressionCV":
+        SklearnLinearModelAssembler,
+    "sklearn.linear_model.passive_aggressive.PassiveAggressiveClassifier":
+        SklearnLinearModelAssembler,
+    "sklearn.linear_model.perceptron.Perceptron":
+        SklearnLinearModelAssembler,
+    "sklearn.linear_model.ridge.RidgeClassifier":
+        SklearnLinearModelAssembler,
+    "sklearn.linear_model.ridge.RidgeClassifierCV":
+        SklearnLinearModelAssembler,
+    "sklearn.linear_model.stochastic_gradient.SGDClassifier":
+        SklearnLinearModelAssembler,
+
+    # Lightning Linear Classifiers
+    "lightning.impl.adagrad.AdaGradClassifier": SklearnLinearModelAssembler,
+    "lightning.impl.primal_cd.CDClassifier": SklearnLinearModelAssembler,
+    "lightning.impl.fista.FistaClassifier": SklearnLinearModelAssembler,
+    "lightning.impl.sag.SAGAClassifier": SklearnLinearModelAssembler,
+    "lightning.impl.sag.SAGClassifier": SklearnLinearModelAssembler,
+    "lightning.impl.sdca.SDCAClassifier": SklearnLinearModelAssembler,
+    "lightning.impl.sgd.SGDClassifier": SklearnLinearModelAssembler,
 
     # Decision trees
-    "DecisionTreeClassifier": TreeModelAssembler,
-    "DecisionTreeRegressor": TreeModelAssembler,
-    "ExtraTreeClassifier": TreeModelAssembler,
-    "ExtraTreeRegressor": TreeModelAssembler,
+    "sklearn.tree.tree.DecisionTreeClassifier": TreeModelAssembler,
+    "sklearn.tree.tree.DecisionTreeRegressor": TreeModelAssembler,
+    "sklearn.tree.tree.ExtraTreeClassifier": TreeModelAssembler,
+    "sklearn.tree.tree.ExtraTreeRegressor": TreeModelAssembler,
 
     # Ensembles
-    "ExtraTreesClassifier": RandomForestModelAssembler,
-    "ExtraTreesRegressor": RandomForestModelAssembler,
-    "RandomForestClassifier": RandomForestModelAssembler,
-    "RandomForestRegressor": RandomForestModelAssembler,
+    "sklearn.ensemble.forest.ExtraTreesClassifier":
+        RandomForestModelAssembler,
+    "sklearn.ensemble.forest.ExtraTreesRegressor":
+        RandomForestModelAssembler,
+    "sklearn.ensemble.forest.RandomForestClassifier":
+        RandomForestModelAssembler,
+    "sklearn.ensemble.forest.RandomForestRegressor":
+        RandomForestModelAssembler,
 }
 
 
+def _get_full_model_name(model):
+    type_name = type(model)
+    return "{}.{}".format(type_name.__module__,
```

Comment:
Not sure, maybe we should take only the first part of `__module__` to be more robust for internal changes in libraries?.. Like `sklearn.DecisionTreeClassifier` instead of `sklearn.tree.tree.DecisionTreeClassifier` and `lightning.CDClassifier` instead of `lightning.impl.primal_cd.CDClassifier`.

Reply:
I think I like this idea very much. Taking the first part will indeed help us to be independent from library internals. The only risk I'm seeing here is a name collision within the same library, which I hope is not something that happens often in the Python world. I'd only suggest to use a separator other than `.` like eg. `_` or `-` so that identifiers are not confused with actual package names. Eg. `sklearn.LogisticRegression` vs `sklearn_LogisticRegression`. Great idea otherwise 👍

Reply:
Thanks a lot for your opinion! I absolutely agree with you about possible collisions. Made update in the latest commit.